File: bonus/find_the_divisibility_array_of_a_string.java
public int[] divisibilityArray(String word, int m)
{
    int n = word.length();
    int[] div = new int[n];
    long curr = 0;

    for (int pos = 0; pos < n; pos++)
    {
        curr = (curr * 10 + (word.charAt(pos) - '0')) % m;

        if (curr == 0)
        {
            div[pos] = 1;
        }
        else
        {
            div[pos] = 0;
        }
    }

    return div;
}

File: bonus/find_the_divisibility_array_of_a_string.js
/**
 * @param {string} word
 * @param {number} m
 * @returns {number[]}
 */
var divisibilityArray = function(word, m)
{
    let div = [];
    let curr = 0;

    for (let pos = 0; pos < word.length; pos++)
    {
        curr = (curr * 10 + +word[pos]) % m;

        if (curr == 0)
        {
            div.push(1);
        }
        else
        {
            div.push(0);
        }
    }

    return div;
};
